beacon_node/network/src/beacon_processor/tests.rs

```diff
@@ -874,3 +889,53 @@ async fn test_rpc_block_reprocessing() {
     // cache handle was dropped.
     assert_eq!(next_block_root, rig.head_root());
 }
+
+/// Ensure that backfill batches get rate-limited and processing is scheduled at specified intervals.
+#[tokio::test]
+async fn test_backfill_sync_processing() {
```

Comment:
TODO (jimmy): this test doesn't really work, the clock manipulation doesn't work as I expected, looks like I have understood some of the test utils in correctly - I'm looking into this

Reply:
Test updated